lib/matplotlib/tests/test_pyplot.py

```diff
@@ -367,3 +368,26 @@ def test_set_current_axes_on_subfigure():
     assert plt.gca() != ax
     plt.sca(ax)
     assert plt.gca() == ax
+
+
+def test_pylab_intergation():
+    pytest.importorskip("IPython")
+    subprocess.run(
+        [
+            sys.executable,
+            "-m",
+            "IPython",
+            "--pylab",
+            "-c",
+            ";".join((
+                "import matplotlib.pyplot as plt",
+                "assert plt._REPL_DISPLAYHOOK == plt._ReplDisplayHook.IPYTHON",
```

Comment:
Actually I don't think this will fail the subprocess if the assertion fails (that'll just print a traceback in ipython)?  I'm not actually even sure of how to exit ipython with nonzero status code (sys.exit is specifically caught by ipython, for example).